### PR TITLE
Bugfix FXIOS-9529 Fix Sentry errors for newly-created iPad windows

### PR DESCRIPTION
**Pull request by mattreaganmozilla**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-9529)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/21089)

## :bulb: Description

The client didn't previously have a notion of a new window (with 0 tabs) vs. a window being restored (also with 0 tabs, but with serialized tabs on-disk). This led to incorrect errors logged to Sentry when creating brand-new iPad windows. Summary:

- On a fresh install of iPad or iPhone, `restoreTabs` is not run; instead `migrateAndRestore()` is called which does not lead to the incorrect log error from `fetchWindowData`
- In all other scenarios (cold launch, or BVC appearance), `restoreTabs()` is called; this includes newly-created windows on iPad
- When a new window is created, the WindowManager knows whether it's new or not but this information wasn't available at the tab manager level and so the tab manager would try to restore tabs anyway even though a brand new window will never have tabs on disk
- The proposed fix is to inject enough information from the WindowManager (which is responsible for providing the UUIDs) down to the tab manager so that it can reason properly about whether it can safely skip `fetchWindowData`

### Notes

The `windowIsNew` state remains `true` during the app lifecycle when the window was created. This shouldn't cause any problems, because AFAICS (and I corroborated this in chat) we only perform one `restoreTabs()`, and once the `tabs` array is populated subsequent calls do nothing (and `tabs` should never go empty). However, if there is some scenario in which we needed to re-restore the window's tabs again in the same lifecycle that it was first created, then `windowIsNew` would need to be set to false at some point otherwise follow-up tab restoration won't do anything. But again, based on everything I can see so far this is not a scenario that should ever occur. So having this state/flag on the window's tab manager should be fine based on all of the information I have.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/Client/TabManagement/TabManagerImplementation.swift [hunk 1]
```diff
@@ -105,7 +107,9 @@ class TabManagerImplementation: LegacyTabManager, Notifiable, WindowSimpleTabsPr
     private func restoreOnly() {
         tabs = [Tab]()
         Task {
-            await buildTabRestore(window: await self.tabDataStore.fetchWindowData(uuid: windowUUID))
+            // Only attempt a tab data store fetch if we know we should have tabs on disk (ignore new windows)
+            let windowData: WindowData? = windowIsNew ? nil : await self.tabDataStore.fetchWindowData(uuid: windowUUID)
```
Comment (mattreaganmozilla) [inline firefox-ios/Client/TabManagement/TabManagerImplementation.swift:111]: Note: this is the key change in the PR.

**Comment by mattreaganmozilla** [comment]
Added a note in the PR description I'd love for extra 👀  on but otherwise I think this should be ready for review.

**Review by adudenamedruby — APPROVED** [review]
LGTM

**Review by OrlaM — APPROVED** [review]

**Comment by mattreaganmozilla** [comment]
Since this is only v130 for now I'm going to merge so I can start resolving the conflicts that will come up with https://github.com/mozilla-mobile/firefox-ios/pull/21183

**Comment by adudenamedruby** [comment]
@Mergify backport release/v128 release/v129

**Pull request merged**